### ISSUE TITLE
BatchNormalization's axis=-1 default isn't foolproof.

### ISSUE THREAD

**Issue by minkooseo** [fchollet/keras#4906]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [V] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [V] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [V] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [V] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hi.

I have a suggestion on BatchNormalization's axis parameter. The parameter has default value -1. For image task, it makes sense only if image dim ordering is tf. 

This can cause trouble for novice users. Every other layer like Dense, Activation, Convolution2D, etc. works magically just after setting 'th' in keras.json. However, BatchNormalization does not. It needs explicit setting of axis. See, as an example, [this question on a MOOC course](http://forums.fast.ai/t/memory-error-issue/228/2) which is about the axis default value.

My proposal is as follows: If not set explicitly, have axis parameter rely on dim_ordering in keras.json just in the same way we set dim_ordering for Convolution2D. As an example, we can change the default axis value to None and then look up json file's dim ordering if it's not set explicitly.

Thanks
Minkoo
